### PR TITLE
Reintroducing React 17 Update

### PR DESCRIPTION
**Pull request by stevematney**
This also adds a fix for our previous issues with deployments, and a canary release workflow to more easily test future changes.

### CODE REVIEW

**Comment by stevematney** [comment]
[This range of commits](https://github.com/WTW-IM/es-components/pull/550/files/4025298715401d193042f9de175c73f8f6718b63..007e5851bf7d88080c7f89d3c12b019ed4351d8c) represents the changes that were added after re-adding the React 17 updates from #535.

**Comment by stevematney** [comment]
I'm going to deploy Homepage and Participant Guidance to dev pointing to the latest canary release so we can check deployed code, but I've tested that release (specifically the bundle) locally and it does work.

The current changes can be tested using: https://unpkg.com/es-components@canary/bundle/main.min.js

**Comment by stevematney** [comment]
> I'm going to deploy Homepage and Participant Guidance to dev pointing to the latest canary release so we can check deployed code...

This deploy was successful, and both micro frontends are working with the canary build.

File: .github/workflows/canary-publish.yml
Comment (aabenoja) [inline .github/workflows/canary-publish.yml:5]: I'm starting to think it'd be good to have a canary branch to auto deploy from anyway.

**Review by Darrken — APPROVED** [review]

**Comment by aabenoja** [comment]
@Darrken @stevematney I think we could use some control over when that canary workflow is triggered. Canary releases are incredibly useful, but I feel like we should have some level of restriction in place to make sure not everyone who has write access to this repo can constantly create canary releases.

**Comment by stevematney** [comment]
> @Darrken @stevematney I think we could use some control over when that canary workflow is triggered. Canary releases are incredibly useful, but I feel like we should have some level of restriction in place to make sure not everyone who has write access to this repo can constantly create canary releases.

I'll have to figure it out, but we could limit it to code owners.

File: .github/workflows/canary-publish.yml
Comment (aabenoja) [inline .github/workflows/canary-publish.yml:9]: Add `environment: canary` here. This you, Ken, Jake and I to approve canary deployments before they go out.

**Comment by aabenoja** [comment]
I've create a `canary` environment for our github workflows. Applying `environment: canary` to the job's properties like above will require codeowner approval before the deployment can continue.

File: .github/workflows/canary-publish.yml
Comment (stevematney) [reply .github/workflows/canary-publish.yml:5]: We could do that as well! I don't think it hurts to have an ad-hoc deploy available for this, but if we had a branch that was always auto-deployed, I don't think that would be bad.

Sometimes we'll have more than one PR being reviewed, and I think having this workflow manually dispatchable will make it easier to test those things as we're reviewing them.

Comment (stevematney) [reply .github/workflows/canary-publish.yml:5]: Another option would be to have a workflow that creates a tag for the branch, and then we could test based on branch tags. (So like `npm install es-components@hot-new-feature-branch`.)

File: .github/workflows/canary-publish.yml
Comment (stevematney) [reply .github/workflows/canary-publish.yml:9]: I think what I've added is correct here. Unfortunately we can't check until this is merged. 😆

**Pull request merged**
